example/cajita_tutorial/05_fast_fourier_transform/fast_fourier_transform_example.cpp

```diff
@@ -0,0 +1,208 @@
+       it would be the default
+    */
+    using HostBackendType = Cajita::Experimental::FFTBackendFFTW;
+
+    double cell_size = 0.1;
+    std::array<bool, 3> is_dim_periodic = { true, true, true };
+    std::array<double, 3> global_low_corner = { -1.0, -2.0, -1.0 };
+    std::array<double, 3> global_high_corner = { 1.0, 1.0, 0.5 };
+    auto global_mesh = Cajita::createUniformGlobalMesh(
+        global_low_corner, global_high_corner, cell_size );
+
+    // Create the global grid.
+    Cajita::DimBlockPartitioner<3> partitioner;
+    auto global_grid = Cajita::createGlobalGrid( MPI_COMM_WORLD, global_mesh,
+                                                 is_dim_periodic, partitioner );
+
+    // Create a local grid.
+    auto local_grid = Cajita::createLocalGrid( global_grid, 0 );
+    auto owned_space = local_grid->indexSpace( Cajita::Own(), Cajita::Cell(),
+                                               Cajita::Local() );
+    auto ghosted_space = local_grid->indexSpace(
+        Cajita::Ghost(), Cajita::Cell(), Cajita::Local() );
+
+    // Create a random vector to transform.
+    // This is the vector of complex data on which we will perform FFTs
+    // Note that for every position there are two values. These correspond
+    // to the real (index 0) and imaginary (index 1) parts
+    auto vector_layout =
+        Cajita::createArrayLayout( local_grid, 2, Cajita::Cell() );
+    auto lhs = Cajita::createArray<double, DeviceType>( "lhs", vector_layout );
+    auto lhs_view = lhs->view();
+    auto lhs_host =
+        Cajita::createArray<double, typename decltype( lhs_view )::array_layout,
+                            Kokkos::HostSpace>( "lhs_host", vector_layout );
+    auto lhs_host_view = Kokkos::create_mirror_view( lhs_view );
+    uint64_t seed =
+        global_grid->blockId() + ( 19383747 % ( global_grid->blockId() + 1 ) );
+    using rnd_type = Kokkos::Random_XorShift64_Pool<Kokkos::HostSpace>;
+    rnd_type pool;
+    pool.init( seed, ghosted_space.size() );
+    for ( int i = owned_space.min( Cajita::Dim::I );
+          i < owned_space.max( Cajita::Dim::I ); ++i )
+        for ( int j = owned_space.min( Cajita::Dim::J );
+              j < owned_space.max( Cajita::Dim::J ); ++j )
+            for ( int k = owned_space.min( Cajita::Dim::K );
+                  k < owned_space.max( Cajita::Dim::K ); ++k )
+            {
+                auto rand = pool.get_state( i + j + k );
+                lhs_host_view( i, j, k, 0 ) =
+                    Kokkos::rand<decltype( rand ), double>::draw( rand, 0.0,
+                                                                  1.0 );
+                lhs_host_view( i, j, k, 1 ) =
+                    Kokkos::rand<decltype( rand ), double>::draw( rand, 0.0,
+                                                                  1.0 );
+            }
+
+    // Copy to the device.
+    Kokkos::deep_copy( lhs_view, lhs_host_view );
```

Comment:
You can just comment that a deep_copy would be needed if we were using a device instead (or use a mirror)

Reply:
I changed this to add a comment, but I think you were actually saying to get rid of the deep_copy?

Reply:
Well you're explicitly running on the host, so I would either mirror or just add a comment saying this would need to be done on the device or copied there if it were intended for a device